File: app/src/main/java/xyz/klinker/messenger/util/billing/ProductType.java
package xyz.klinker.messenger.util.billing;

public enum ProductType {
    SINGLE_PURCHASE("inapp"), SUBSCRIPTION("subs");

    private String identifier;

    ProductType(String identifier) {
        this.identifier = identifier;
    }

    public String getIdentifier() {
        return identifier;
    }

    public String[] getAvailableProductIds() {
        if (this == SINGLE_PURCHASE) {
            return new String[] {
                    "lifetime"
            };
        } else {
            return new String[] {
                    "subscription_one_month",
                    "subscription_three_months",
                    "subscription_one_year"
            };
        }
    }
}

File: shared/src/main/java/xyz/klinker/messenger/shared/util/billing/ProductType.kt
package xyz.klinker.messenger.shared.util.billing

enum class ProductType private constructor(val identifier: String) {
    SINGLE_PURCHASE("inapp"), SUBSCRIPTION("subs");

    val availableProductIds: Array<String>
        get() = if (this == SINGLE_PURCHASE) {
            arrayOf("lifetime2")
        } else {
            arrayOf("subscription_one_month_no_trial", "subscription_three_months_no_trial", "subscription_one_year_no_trial")
        }
}
